Bindings/Packages/com.virtualmaker.bindings/Runtime/Bindings.InputActions.cs

```diff
@@ -0,0 +1,190 @@
+#if UNITY_INPUT_SYSTEM
+using System;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace VirtualMaker.Bindings
+{
+    public partial class Bindings
+    {
+        public void Bind<T>(InputActionReference inputActionReference,
+            Property<T> performedProp, Property<T> startedProp = null, Property<T> canceledProp = null,
+            Action<T> performedCb = null, Action<T> startedCb = null, Action<T> canceledCb = null)
+            where T : struct
+        {
+            if (performedProp != null)
+            {
+                performedCb?.Invoke(performedProp.Value);
+            }
+
+            if (startedProp != null)
+            {
+                startedCb?.Invoke(startedProp.Value);
+            }
+
+            if (canceledProp != null)
+            {
+                canceledCb?.Invoke(canceledProp.Value);
+            }
+
+            BindDeferred(inputActionReference,
+                performedProp, startedProp, canceledProp,
+                performedCb, startedCb, canceledCb);
+        }
+
+        public void BindDeferred<T>(InputActionReference inputActionReference,
+            Property<T> performedProp, Property<T> startedProp = null, Property<T> canceledProp = null,
+            Action<T> performedCb = null, Action<T> startedCb = null, Action<T> canceledCb = null)
+            where T : struct
+        {
+            void Performed(T value)
+            {
+                if (performedProp != null)
+                {
+                    performedProp.Value = value;
+                    performedCb?.Invoke(performedProp.Value);
+                }
+                else
+                {
+                    performedCb?.Invoke(value);
+                }
+            }
+
+            void Started(T value)
+            {
+                if (startedProp != null)
+                {
+                    startedProp.Value = value;
+                    startedCb?.Invoke(startedProp.Value);
+                }
+                else
+                {
+                    startedCb?.Invoke(value);
+                }
+            }
+
+            void Canceled(T value)
```

Comment:
what does the value here mean?

Reply:
since the T can be context of input action, it is staying as is.